test/unit/com/unifina/controller/api/IntegrationKeyApiControllerSpec.groovy

```diff
@@ -20,6 +20,8 @@ class IntegrationKeyApiControllerSpec extends ControllerSpecification {
 	SecUser someoneElse
 
 	def setup() {
+		EthereumIntegrationKeyService ethereumIntegrationKeyService = new EthereumIntegrationKeyService()
```

Comment:
- Shadows the class field with the same name `ethereumIntegrationKeyService`
- Should prefer mocks in unit tests instead of real collaborators

Which test case fails without this? 

Looks like in this test suite, the `EthereumIntegrationKeyService` is commonly mocked locally in each test cases, e.g. like in line 123. Not sure why this repetition is there. The mock could also be registered for all tests in `setup` with:

```
ethereumIntegrationKeyService = mockBean(EthereumIntegrationKeyService, Mock(EthereumIntegrationKeyService))
```

One of the tests use `Stub` instead of `Mock`, not sure why. Maybe this is the reason for the repetition (one test case needs a different setup).

Reply:
`ethereumIntegrationKeyService = mockBean(EthereumIntegrationKeyService, Mock(EthereumIntegrationKeyService))` works instead.